clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1357,7 +1367,11 @@ private void close(Duration timeout, boolean swallowException) {
         swallow(log, Level.ERROR, "Failed invoking asynchronous commit callbacks while closing consumer",
```

Comment:
github won't let me leave a comment on this line but look at the call directly above here -- wondering does it make more sense to integrate the `membershipOperation` into the ``#leaveGroupOnClose` method?